src/lib/logic/PeopleComparator.ts

```diff
@@ -0,0 +1,46 @@
+
+import { MatrixResult } from "$lib/types/MatrixResult";
+import type { Person } from "../types/Person";
+import { AspectCalculator } from "./AspectCalculator";
+import { HouseCalculator } from "./HouseCalculator";
+
+export class PeopleComparator {
+    private readonly aspectCalculator = new AspectCalculator();
+    private readonly houseCalculator = new HouseCalculator();
+
+    /**
+     * Compares the people and leaves the results in a matrix.
+     * @param people the people to compare among themselves.
+     * @returns a symmetric matrix with the upper half filled with the results
+     * of the comparissons.
+     */
+    public comparePeople(people: Array<Person>): MatrixResult[][] {
+        const resultsMatrix: MatrixResult[][] = [];
+
+        for (let i = 0; i < people.length; i++) {
+            // Initialize the rows for each column.
+            resultsMatrix[i] = [];
+
+            for (let j = 0; j < people.length; j++) {
+                // The diagonal of the matrix represents as if we were
+                // one person to himself, and that doesn't make sense.
+                // Also,the matrix is symetrical, so it doesn't make
+                // sense to compare one person to another and back.
+                if (i === j || i > j) {
+                    continue;
+                }
```

Comment:
```suggestion
            for (let j = i + 1; j < people.length; j++) {
                // The diagonal of the matrix represents as if we were
                // one person to himself, and that doesn't make sense.
                // Also,the matrix is symetrical, so it doesn't make
                // sense to compare one person to another and back.
```

Reply:
Could be simplified

Reply:
ah, hold on- I'm using the other side of the matrix :)

Reply:
I think this is equivalent to the lower side of the matrix, but i prefer the for loop with the upper side :sweat_smile: 

```suggestion
            for (let j = 0; j < i; j++) {
                // The diagonal of the matrix represents as if we were
                // one person to himself, and that doesn't make sense.
                // Also,the matrix is symetrical, so it doesn't make
                // sense to compare one person to another and back.
```